src/client/js/dialog.js

```diff
@@ -29,9 +29,11 @@ function handleEvent (e) {
     case e.target.matches('a[href*="dialog/"]'):
       e.preventDefault()
       openDialog(e.target.href)
+      window.gtag('event', 'Dialog', { action: 'open', page_location: location.href })
       break
     case e.target.matches('button[data-dialog]'):
       openDialog(`dialog/${e.target.dataset.dialog}`)
+      window.gtag('event', 'Dialog', { action: 'open', page_location: location.href })
```

Comment:
Should we just put this call once in the `openDialog()` function below, rather than twice in the switch?  

Additionally, `openDialog()` is called with a path that might be helpful to include – or even `partialName` derived from that path.  This way we can know exactly what dialog was opened, instead of the containing page which could have multiple dialogs.

If you _do_ opt for this suggestion, the gtag should be called somewhere after the line `if (!res.ok) throw new Error('Bad fetch response')`, which signifies the modal was open successfully.  You could still add a gtag before there, or specifically in the event of an error 🤷

Reply:
Yeah after reading the docs and the UA implementation in the previous version of Monitor, I think that setting success/fail parameters in the events is probably useful anyway. Some of Tony's ask around this included that so we might as well try to be consistent.

Reply:
Well, hm. Actually this wouldn't fail for any reason that is the user's fault... I think just checking for dialog open is probably fine. The line between GA and Sentry can be strange :)